src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -431,8 +431,9 @@ public synchronized boolean run(String[] args) {
                     file = new File(fname);
                 } else {
                     file = createTemporaryFile("tmpJar", ".jar");
-                    try (InputStream in = new FileInputStream(FileDescriptor.in)) {
-                        Files.copy(in, file.toPath());
```

Comment:
Did you try adding the REPLACE_EXISTING option to Files.copy, I assume that will fix it.

Reply:
Hello Alan, I had thought about that, but then I looked at the implementation of `Files.copy(...)` with `REPLACE_EXISTING`. If that option is specified, the `Files.copy(...)` implementation first deletes the existing file:
```
// attempt to delete an existing file
if (replaceExisting) {
    deleteIfExists(target);
}
```
before it initiates the copying. It didn't feel right to be explicitly creating a file (before the call to Files.copy) and then having it deleted due to the use of  `REPLACE_EXISTING`. So I went ahead with this alternate approach.

Reply:
Thank you Lance for the review. Alan, is it OK to proceed with this current change or do you think we should pursue the `REPLACE_EXISTING` option here?

Reply:
I don't object to what you have, it's the mixing of old and new APIs that jumped out. Maybe some day there will be some wider updates to the jar tool in this area, e.g. it could have use a temp directory rather than a temp file.

Reply:
Thank you for that input, Alan. I'll go ahead with integrating this current PR. As a separate activity, I will take a broader look at this jar tool code and see what changes can be accomodated to modernize the code.